### PR TITLE
Add validation splits to the food101 dataset

### LINKED ISSUE #1215: There is only a train split in the food101 dataset.
**Short description**

There is only a train split in the [food101 dataset](https://www.tensorflow.org/datasets/catalog/food101). In the [source data](https://www.vision.ee.ethz.ch/datasets_extra/food-101/), the dataset consists of the following configurations: 

```
  num_images = {
    'train': 75750,
    'validation': 25250,
  }
```

I think it should be fixed.
Thanks

### PR DESCRIPTION
**Pull request by cking100**
Food101 dataset of TensorFlow Datasets is having only the train split of the dataset. However, the original source of the dataset contains 25250 validation data which are being completely ignored in TFDS.

Fixes #1215

### CODE REVIEW

**Comment by cking100** [comment]
@googlebot I signed it!

**Comment by cking100** [comment]
> Please also add exact number of some fake_examples here. Similar to the train. Thanks.

Hey, @ChanchalKumarMaji. I have added the examples as you said and the tests cases seem to be working fine. But, I'm getting an error while running `ds_val = tfds.load(name="food101", split="validate", shuffle_files=True)` like validation directory not found. I think this is due to original dataset not divided into train and testing data. The dataset has files like `test.json` and `train.json`. So, do I have to use them to split data. if yes, then can you refer how. Thanks

**Comment by cking100** [comment]
@ChanchalKumarMaji Could you please look at it now? Thanks

File: tensorflow_datasets/image/food101.py [hunk 1]
```diff
@@ -69,26 +70,38 @@ def _info(self):
   def _split_generators(self, dl_manager):
     """Define Splits."""
 
-    path = dl_manager.download_and_extract(_BASE_URL)
+    dl_path = dl_manager.download_and_extract(_BASE_URL)
+    path = os.path.join(os.path.join(dl_path, "food-101"), "meta")
```
Comment (captain-pool) [inline tensorflow_datasets/image/food101.py]: You can replace this with `os.path.join(dl_path, "food-101", "meta")`

File: tensorflow_datasets/image/food101.py [hunk 2]
```diff
@@ -69,26 +70,38 @@ def _info(self):
...
-      for image_name in tf.io.gfile.listdir(class_dir_path):
-        image = os.path.join(class_dir_path, image_name)
-        yield image, {
-            "image": image,
-            "label": class_name,
-        }
+    data_dir_path = os.path.join(os.path.dirname(os.path.dirname(path)), 'images')
+    with tf.io.gfile.GFile(path) as f:
+        for line in f:
+            data = json.loads(line)
```
Comment (captain-pool) [inline tensorflow_datasets/image/food101.py]: Is this supposed to be like this? I mean, it looks like `data` will only contain the last line after all iterations are completed. And that last line will only be processed. Not sure how that's gonna work properly.

Maybe you meant this:
```python3
with tf.io.gfile.GFile(path) as f:
  data = json.loads(f.read())
. . . . .
```

Comment (cking100) [reply tensorflow_datasets/image/food101.py]: It doesn't make any difference as there is only a single line.  But, I would have to agree that the code given by you will take diff lines into consideration as well. I will update it.

**Comment by cking100** [comment]
@captain-pool Made the requested changes. Please take a look. Thanks

**Comment by captain-pool** [comment]
CC: @rsepassi @Conchylicultor @cyfra

**Comment by ChanchalKumarMaji** [comment]
Many changes are still required:
1. Indentation Errors are there.
2. The `fake_samples` should be made and not taken from somewhere.

**Comment by cking100** [comment]
@ChanchalKumarMaji Just to clarify, the indentation in `food101.py` file right?

**Comment by captain-pool** [comment]
@cking100 you need to generate random fake samples.

**Comment by cking100** [comment]
> @cking100 you need to generate random fake samples.

How can I do that? Could you help me. Is there any online tool or something to generate random image samples? @captain-pool

**Comment by cyfra** [comment]
Thanks @cking100. Code looks ok - the only thing missing is replacing your pictures with fake ones.

You can use the functions from testing/fake_data_utils.py.

**Comment by cking100** [comment]
@cyfra Thanks a lot for your help. I have replaced the pictures with the fake ones.

File: tensorflow_datasets/image/food101.py [hunk 3]
```diff
@@ -69,26 +70,37 @@ def _info(self):
   def _split_generators(self, dl_manager):
     """Define Splits."""
 
-    path = dl_manager.download_and_extract(_BASE_URL)
+    dl_path = dl_manager.download_and_extract(_BASE_URL)
```
Comment (cyfra) [inline tensorflow_datasets/image/food101.py]: Please also change the version of the dataset to 2.0.0
(in line 53)

File: tensorflow_datasets/image/food101.py [hunk 4]
```diff
@@ -69,26 +70,37 @@ def _info(self):
...
+    path = os.path.join(dl_path, "food-101", "meta")
 
     return [
         tfds.core.SplitGenerator(
             name=tfds.Split.TRAIN,
             num_shards=4,
```
Comment (cyfra) [inline tensorflow_datasets/image/food101.py]: num_Shards can be removed

**Comment by cking100** [comment]
I have made the requested changes.

**Pull request merged**